File: ai-models/services/weather_service.py
import requests

API_KEY = "692dab038a2870bc0160d7767f889d5f"
CITY = "Guwahati"

def get_weather():

    url = (
        f"https://api.openweathermap.org/data/2.5/weather"
        f"?q={CITY}"
        f"&appid={API_KEY}"
        f"&units=metric"
    )

    response = requests.get(url)

    print("Status Code:", response.status_code)
    print("Response:", response.text)

    data = response.json()

    return {
        "temperature": data["main"]["temp"],
        "humidity": data["main"]["humidity"],
        "rainfall": data.get("rain", {}).get("1h", 0)
    }

File: backend/services/weather.service.ts
import axios from "axios";

export async function getWeather() {

    const apiKey =
        process.env.OPENWEATHER_API_KEY;

    const city =
        process.env.OPENWEATHER_CITY;

    const url =
        `https://api.openweathermap.org/data/2.5/weather?q=${city},IN&appid=${apiKey}&units=metric`;

    const response =
        await axios.get(url);

    const data = response.data;

    return {

        temperature:
            data.main.temp,

        humidity:
            data.main.humidity,

        wind:
            data.wind.speed,

        rainfall:
            data.rain?.["1h"] || 0,

        condition:
            data.weather[0].main
    };
}